bot/exts/moderation/watchchannels/talentpool.py

```diff
@@ -176,13 +175,39 @@ async def nomination_edit_group(self, ctx: Context) -> None:
 
     @nomination_edit_group.command(name='reason')
     @has_any_role(*MODERATION_ROLES)
-    async def edit_reason_command(self, ctx: Context, nomination_id: int, *, reason: str) -> None:
-        """
-        Edits the reason/unnominate reason for the nomination with the given `id` depending on the status.
+    async def edit_reason_command(self, ctx: Context, nomination_id: int, actor: FetchedMember, *, reason: str) -> None:
+        """Edits the reason of `actor` entry for the nomination with the given `id`."""
+        try:
+            nomination = await self.bot.api_client.get(f"{self.api_endpoint}/{nomination_id}")
+        except ResponseCodeError as e:
+            if e.response.status == 404:
+                self.log.trace(f"Nomination API 404: Can't nomination with id {nomination_id}")
+                await ctx.send(f":x: Can't find a nomination with id `{nomination_id}`")
+                return
+            else:
+                raise
 
-        If the nomination is active, the reason for nominating the user will be edited;
-        If the nomination is no longer active, the reason for ending the nomination will be edited instead.
-        """
+        if not nomination["active"]:
+            await ctx.send(":x: Can't edit reason of ended nomination.")
+            return
+
+        if not any(entry["actor"] == actor.id for entry in nomination["entries"]):
+            await ctx.send(f":x: {actor} don't have entry for this nomination.")
+            return
+
+        self.log.trace(f"Changing reason for nomination with id {nomination_id} of actor {actor} to {reason}")
```

Comment:
I'd use the repr of the reason here
```suggestion
        self.log.trace(f"Changing reason for nomination with id {nomination_id} of actor {actor} to {repr(reason)}")
```